WebViewControl/AssemblyCache.cs

```diff
@@ -8,21 +8,31 @@ namespace WebViewControl {
      internal class AssemblyCache {
 
         private object SyncRoot { get; } = new object();
-        private Dictionary<string, Assembly> assemblies;
+
+        // We now allow to load multiple versions of the same assembly, which means that resource urls can
+        // (optionally) specify the version. We don't force the version to be specified to maintain backwards
+        // compatibility, and thus need to store assemblies by their name, in addition to by (name,version).
+        // Note that assembliesByName naturally doesn't handle duplicates, and no guarantee is provided about
+        // which version is resolved if there are multiple loaded assemblies with the same name.
+        // This, consumer apps are encouraged to include the version in the resource url
+        private Dictionary<(string, Version), Assembly> assembliesByNameAndVersion;
```

Comment:
you can still have just one dictionary will mantaining back-compat, storing a null version when no version should be used  
```suggestion
        private Dictionary<(string, Version?), Assembly> assembliesByNameAndVersion;
```

Reply:
another approach could be using a list of versions for the same dll (considering that most of the times there should exist a few, 1 or 2) and then... you could opt for always loading the most recent one